src/lib.rs

```diff
@@ -315,6 +316,28 @@ pub trait SharedLibrary: Sized + Debug {
     /// Get the debug-id of this shared library if available.
     fn id(&self) -> Option<SharedLibraryId>;
 
+    /// Returns the address of where the library is loaded.
+    ///
+    /// This typically is the stated virtual memory address of the first
+    /// code segment.
+    fn stated_virtual_memory_address(&self) -> Svma {
+        self.segments()
+            .find(|x| x.is_code())
+            .map(|x| x.stated_virtual_memory_address())
+            .unwrap_or_else(|| Svma(ptr::null()))
```

Comment:
So the main goal is to mirror what breakpad and other server side symbolication systems consider the "load address" of the image. The current implementation is definitely not correct based on the earlier comments. I will most likely implement this manually per platform.

The goal would be to achieve the image address that matches what breakpad does because this is an established behavior that systems like Sentry and others use.

Reply:
Also to clarify: the null case should really never happen (tm).

Reply:
> Also to clarify: the null case should really never happen (tm).

If that is the case, then I think `Svma(usize::MAX as _)` would be safer, since its potential address range is a single byte.

But I'll defer until you figure out the other issues you mentioned.